handbook/engineering/distribution/observability/monitoring_architecture.md

```diff
@@ -0,0 +1,114 @@
+* Alert and recording rules generated by the [monitoring generator](#monitoring-generator).
+* [Alertmanager](https://prometheus.io/docs/alerting/latest/alertmanager/), which handles alerts from Prometheus.
+* [prom-wrapper](https://github.com/sourcegraph/sourcegraph/tree/master/docker-images/prometheus/cmd/prom-wrapper), which subscribes to updates in [site configuration](https://docs.sourcegraph.com/admin/config/site_config) and propagates relevant settings to Alertmanager configuration.
+
+#### Alert count metrics
+
+`alert_count` metrics are special Prometheus recording rules that evaluate a single upper or lower bound, as defined in an [Observable](https://sourcegraph.com/search?q=repo:%5Egithub%5C.com/sourcegraph/sourcegraph%24+file:generator.go+type+Observable+struct+%7B:%5Bdef%5D%7D&patternType=structural) and generated by the [monitoring generator](#monitoring-generator). This metric is always either 0 if the threshold is not exceeded (or data does not exist, if configured), or 1 if the threshold is exceeded. This allows historical alert data to easily be [consumed programmatically](https://docs.sourcegraph.com/admin/observability/alerting_custom_consumption).
+
+Learn more about the `alert_count` metrics in the [metrics guide](https://docs.sourcegraph.com/admin/observability/metrics_guide#alert-count).
+
+*Rationale for `alert_count`*: TODO(@slimsag)
+
+#### Alert notifications
+
+We use [Alertmanager](https://prometheus.io/docs/alerting/latest/alertmanager/) for:
+
+- Providing data about currently active Sourcegraph alerts.
+- Routing alerts to appropriate receivers and silencing them when desired, [configured using site configuration](#alert-notifications).
+
+Alertmanager is bundled in `sourcegraph/prometheus`, and notifications are configured for Sourcegraph alerts [using site configuration](https://docs.sourcegraph.com/admin/observability/alerting). This functionality is provided by the [prom-wrapper](#prom-wrapper).
+
+*Rationale for notifiers in site configuration*: Due to the limitations of [admin reverse-proxies](#admin-reverse-proxy), alerts cannot be configured without port-forwarding or custom ConfigMaps, something we [want to avoid](monitoring_pillars.md#long-term-vision).
+
+*Rationale for Alertmanager*: An approach for notifiers using Grafana was considered, but had some issues outlined in [#11832](https://github.com/sourcegraph/sourcegraph/pull/11832), so Alertmanager was selected as our notification provider.
+
+*Rationale for silencing in site configuration*: Similar to the [Grafana admin reverse-proxy](#admin-reverse-proxy), silencing using the Alertmanager UI would require port-forwarding, something we [want to avoid](monitoring_pillars.md#long-term-vision).
+
+#### prom-wrapper
+
+The [prom-wrapper](https://github.com/sourcegraph/sourcegraph/tree/master/docker-images/prometheus/cmd/prom-wrapper) is the entrypoint program for `sourcegraph/prometheus` and it:
+
+* Handles starting up Prometheus and Alertmanager
```

Comment:
I think we should add a note on why we chose to use a wrapper in the same container vs multiple containers. As I recall from the conversation with @slimsag, this was to avoid increasing the container count, but there is no hard requirement, as we could share the file via a `ro:` mount and use the reload API of Prometheus and Alertmanager.

Reply:
Will add a section!

Reply:
8defdc002a3f6229bda6e9edcf1ac7d08d1e9cda